crates/router/src/routes/payments.rs

```diff
@@ -2944,6 +2952,99 @@ pub async fn payment_status(
     .await
 }
 
+#[cfg(feature = "v2")]
+#[instrument(skip(state, req), fields(flow, payment_id))]
+pub async fn payments_status_with_gateway_creds(
+    state: web::Data<app::AppState>,
+    req: actix_web::HttpRequest,
+    payload: web::Json<api_models::payments::PaymentsRetrieveRequest>,
+    path: web::Path<common_utils::id_type::GlobalPaymentId>,
+) -> impl Responder {
+    use hyperswitch_domain_models::payments::PaymentStatusData;
+
+    let flow = match payload.force_sync {
```

Comment:
Nit: Can we record `flow` in the current span as well, considering that we have it included as a `field` in the `#[instrument]` annotation?